src/portable/nordic/nrf5x/dcd_nrf5x.c

```diff
@@ -1059,7 +1062,13 @@ void tusb_hal_nrf_power_event (uint32_t event)
 
       // Enable interrupt, priorities should be set by application
       NVIC_ClearPendingIRQ(USBD_IRQn);
-      NVIC_EnableIRQ(USBD_IRQn);
+      // Don't enable USBD interrupt yet, dcd_init() did not finish yet
+      // Interrupt will be enabled by tud_init(), when USB stack is ready
+      // to handle interrupts.
+      if (_dcd.init_called)
+      {
+        NVIC_EnableIRQ(USBD_IRQn);
+      }
```

Comment:
maybe we should just use the `tud_inited()`. Although dcd shouldn't call higher layer API, i guess it is not that much of dependency. That would be one less variable to worry about.

Reply:
I did missed that, update is coming...

Reply:
Easy way for me to trigger this race (and then verify that solution work) is to have
```c
#define CFG_TUSB_DEBUG              1
```

Reply:
thanks for the hint, in this case, the code proves itself :+1: